src/EditorFeatures/VisualBasicTest/RemoveSharedFromModuleMembers/RemoveSharedFromModuleMembersTests.vb

```diff
@@ -0,0 +1,193 @@
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    Public ReadOnly Property X As Integer
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
+
+        <Fact>
+        Public Async Function TestSharedFullPropertyInModule() As Task
+            Dim source = "
+Public Module M
+    Public {|BC30503:Shared|} Property X As Integer
+        Get
+            Return 0
+        End Get
+        Set(value As Integer)
+        End Set
+    End Property
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    Public Property X As Integer
+        Get
+            Return 0
+        End Get
+        Set(value As Integer)
+        End Set
+    End Property
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
+
+        <Fact>
+        Public Async Function TestSharedFunctionInModule() As Task
+            Dim source = "
+Public Module M
+    Public {|BC30433:Shared|} Function DoSomething()
+    End Function
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    Public Function DoSomething()
+    End Function
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
+
+        <Fact>
+        Public Async Function TestSharedFunctionWithoutAccessModifierInModule() As Task
+            Dim source = "
+Public Module M
+    ' Trivia
+    {|BC30433:Shared|} Function DoSomething()
+    End Function
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    ' Trivia
+    Function DoSomething()
+    End Function
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
+
+        <Fact>
+        Public Async Function TestSharedSubInModule() As Task
+            Dim source = "
+Public Module M
+    Public {|BC30433:Shared|} Sub DoSomething()
+    End Sub
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    Public Sub DoSomething()
+    End Sub
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
+
+        <Fact>
+        Public Async Function TestSharedEventInModule() As Task
+            Dim source = "
+Public Module M
+    {|BC30434:Shared|} Event OnSomething()
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    Event OnSomething()
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
+
+        <Fact>
+        Public Async Function TestFixAll() As Task
+            Dim source = "
+Public Module M
+    Public {|BC30593:Shared|} x As Integer
+
+    Public {|BC30433:Shared|} Sub DoSomething()
+    End Sub
+End Module
+"
+            Dim fixedSource = "
+Public Module M
+    Public x As Integer
+
+    Public Sub DoSomething()
+    End Sub
+End Module
+"
+            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
+        End Function
```

Comment:
do you have a fix-all test?  do you have one with a shared multiple-field declaration?

Reply:
> do you have a fix-all test?

Yes the last test ia a fix all test.

>  do you have one with a shared multiple-field declaration?

Will add one.

Reply:
> do you have one with a shared multiple-field declaration?

Added the test but it fails:

```vb
        <Fact>
        Public Async Function TestSharedFieldWithMultipleVariablesInModule() As Task
            Dim source = "
Public Module M
    {|BC30593:Shared|} x, y As Integer
End Module
"
            Dim fixedSource = "
Public Module M
    Dim x, y As Integer
End Module
"
            Await VerifyVB.VerifyCodeFixAsync(source, fixedSource)
```

Diff:

```diff
 Node(CompilationUnit):
   Node(ModuleBlock):
     Node(ModuleStatement):
       Token(PublicKeyword): Public
       Token(ModuleKeyword): Module
       Token(IdentifierToken): M
-    Node(EmptyStatement):
-      Token(EmptyToken): 
+    Node(FieldDeclaration):
+      Node(VariableDeclarator):
+        Node(ModifiedIdentifier):
+          Token(IdentifierToken): x
+        Token(CommaToken): ,
+        Node(ModifiedIdentifier):
+          Token(IdentifierToken): y
+        Node(SimpleAsClause):
+          Token(AsKeyword): As
+          Node(PredefinedType):
+            Token(IntegerKeyword): Integer
     Node(EndModuleStatement):
       Token(EndKeyword): End
       Token(ModuleKeyword): Module
   Token(EndOfFileToken):
```

This should be a bug in SyntaxGenerator. I'll debug it.